src/core/crypto/ecdsa.hpp

```diff
@@ -146,6 +168,31 @@ class P256
          */
         Error GetPublicKey(PublicKey &aPublicKey) const { return otPlatCryptoEcdsaGetPublicKey(this, &aPublicKey); }
 
+        /**
+         * This method sets the public key from the buffer passed. This assumes the key to be in DER format.
+         *
+         * @param[in] aDerBytes    Buffer with public key in DER format.
+         * @param[in] aDerLength   Length of the buffer passed.
+         *
+         * @retval kErrorNone      Public key was updated successfully.
+         * @retval kErrorFailed    Failed to update the public key.
+         *
+         */
+        Error SetKeyPairFrom(const uint8_t *aDerBytes, uint8_t aDerLength)
+        {
+            Error error = kErrorNone;
+
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+            error = Crypto::Storage::ImportKey(mKeyRef, Storage::kKeyTypeEcdsa, Storage::kKeyAlgorithmEcdsa,
+                                               (Storage::kUsageSignHash | Storage::kUsageVerifyHash),
+                                               Storage::kTypePersistent, aDerBytes, aDerLength);
+#else
+            memcpy(mDerBytes, aDerBytes, aDerLength);
```

Comment:
To be safe add check that given `aDerLength` is smaller sizeof `mDerBytes` array before `memcpy` (avoid write after end of array?)